service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/StatementTimeMappingUtils.java

```diff
@@ -93,4 +96,30 @@ public static String prepareEffectiveTimeForAllergyIntolerance(String onsetDate,
         }
         return DEFAULT_TIME_VALUE;
     }
+
+    public static String prepareEffectiveTimeForMedicationRequest(MedicationRequest medicationRequest) {
+        if (medicationRequest.getDispenseRequest().hasValidityPeriod()
+            && medicationRequest.getDispenseRequest().getValidityPeriod().hasStart()) {
+            Period period = medicationRequest.getDispenseRequest().getValidityPeriod();
+
+            if (medicationRequest.getDispenseRequest().getValidityPeriod().hasEnd()) {
+                return String.format(EFFECTIVE_TIME_FULL_TEMPLATE,
+                    toHl7Format(period.getStartElement()),
+                    toHl7Format(period.getEndElement()));
+            }
+
+            return String.format(EFFECTIVE_TIME_CENTER_TEMPLATE, toHl7Format(
+                medicationRequest.getDispenseRequest().getValidityPeriod().getStartElement()));
+        }
+        throw new EhrMapperException("Could not map Effective Time for Medication Request");
```

Comment:
Some repetition could be avoided:
```suggestion
        final var dispenseRequest = medicationRequest.getDispenseRequest();
        if (dispenseRequest.hasValidityPeriod() && dispenseRequest.getValidityPeriod().hasStart()) {
            Period period = dispenseRequest.getValidityPeriod();
            DateTimeType startElement = period.getStartElement();

            if (period.hasEnd()) {
                return String.format(EFFECTIVE_TIME_FULL_TEMPLATE,
                    toHl7Format(startElement),
                    toHl7Format(period.getEndElement()));
            }

            return String.format(EFFECTIVE_TIME_CENTER_TEMPLATE, toHl7Format(startElement));
        }
        throw new EhrMapperException("Could not map Effective Time for Medication Request");
```